app/actions/websocket/posts.ts

```diff
@@ -203,10 +211,44 @@ export async function handlePostEdited(serverUrl: string, msg: WebSocketMessage)
     }
 }
 
-export function handlePostDeleted(serverUrl: string, msg: WebSocketMessage) {
+export async function handlePostDeleted(serverUrl: string, msg: WebSocketMessage) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return;
+    }
     try {
-        const data: Post = JSON.parse(msg.data.post);
-        markPostAsDeleted(serverUrl, data);
+        const {database} = operator;
+
+        const post: Post = JSON.parse(msg.data.post);
+
+        const models: Model[] = [];
+
+        const {model: deleteModel} = await markPostAsDeleted(serverUrl, post, true);
+        if (deleteModel) {
+            models.push(deleteModel);
+        }
+
+        // update thread when a reply is deleted and CRT is enabled
+        if (post.root_id) {
```

Comment:
Do we have to take care about deleting the root post? Or that is somehow handled at markPostAsDeleted?

Reply:
Yes, existing code already handles it. This code is to handle the `replyCount` and fetch the thread again.